### PR TITLE
Better handle header layout with no mega menu

### PR DESCRIPTION
**Pull request by chosak**
If no mega menus have been defined in Wagtail, the new mega menu implementation (introduced in #5471 via the `MEGA_MENU_BACKEND_V2` feature flag, and now turned on in production) will raise an exception. This is problematic because it prevents us from running Python tests with defining some placeholder menu, and prevents use of cfgov-refresh on an empty database without defining a menu.

To better handle these cases, it would be better if we could instead just fallback to showing no menu if none are defined. Additionally, this opens up the possibility of deliberately hiding the menu on certain pages in an organized way, as opposed to via custom base templates.

This change makes that possible, and makes the necessary frontend code changes to handle the case where the menu is not displayed.

@contolini

## Testing

In order to test this you'll need either a very recent production dump with `MEGA_MENU_BACKEND_V2` already turned on, or you'll need to turn it on locally and run `cfgov/manage.py migrate_menu_content` to create menu content for the new backend (for example [this custom header code](https://github.com/cfpb/teachers-digital-platform/blob/d57248c839b91ac5a89c361d9c6984753effc239/teachers_digital_platform/jinja2/teachers_digital_platform/crt-base.html#L35-L60) for [this live page](https://www.consumerfinance.gov/practitioner-resources/youth-financial-education/curriculum-review/before-you-begin/).

Run `gulp clean && gulp build` to build frontend assets.

First, run a local server and visit http://localhost:8000/ to confirm proper functionality for both desktop and mobile browser sizes.

Next, go to the [menu edit area in Wagtail](http://localhost:8000/admin/mega_menu/menu/), and delete the existing menu data.

Now, revisit http://localhost:8000/ and review the design and functionality when the menu isn't being displayed.

## Screenshots

Expected behavior:

no-menu

## Notes

- @schbetsy if you have time, I've tagged you here since you did such a thorough review on #5471, and you commented on this specific case.

## Todos

- Now that the new mega menu backend is live, there's some additional cleanup that needs to happen, which I'll propose in a future PR.

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

File: cfgov/jinja2/v1/_includes/organisms/header.html [hunk 1]
```diff
@@ -8,17 +8,21 @@
...
-                Default is true.
+   show_banner:     Whether the global banner molecule is included.
+                    Default is false.
 
-   language:    The page's language value (used to display different
-                information on Spanish pages). Defaults to English.
+   show_mega_menu:  Whether the mega menu is included. Default is true.
+
+   language:        The page's language value (used to display different
+                    information on Spanish pages). Defaults to English.
```
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/organisms/header.html]: Could say `Defaults to 'en' for English.`

File: cfgov/unprocessed/js/organisms/Header.js [hunk 2]
```diff
@@ -53,10 +53,13 @@ function Header( element ) {
     _globalSearch.addEventListener( 'expandBegin', _searchExpandBegin );
     _globalSearch.init();
 
-    _megaMenu = new MegaMenu( _dom );
-    _megaMenu.addEventListener( 'rootExpandBegin', _megaMenuExpandBegin );
-    _megaMenu.addEventListener( 'rootCollapseEnd', _megaMenuCollapseEnd );
-    _megaMenu.init();
+    // Don't initialize the mega menu if it isn't on the page.
+    if ( _dom.classList.contains( 'o-header__mega-menu' ) ) {
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/organisms/Header.js]: Update to ``if ( _dom.classList.contains( `${ BASE_CLASS }__mega-menu` ) ) {``

File: cfgov/jinja2/v1/_includes/organisms/header.html (hunk 1 above)
Comment (chosak) [reply cfgov/jinja2/v1/_includes/organisms/header.html]: Fixed in 9bbb731.

File: cfgov/unprocessed/js/organisms/Header.js (hunk 2 above)
Comment (chosak) [reply cfgov/unprocessed/js/organisms/Header.js]: Fixed in 866cd10.

**Comment by chosak** [comment]
> When in mobile and opening the global search, the mega menu is closed here. Without a menu, this throws an error. So we'll either want to check for the menu at that point, or perhaps better, don't add the listener here, when the menu does not exist.

Thanks @anselmbradford. Nice catch. I've fixed in 08a6bfc.

**Comment by chosak** [comment]
@anselmbradford rewrote search fix per your offline advice in 75fbdb6.

File: cfgov/unprocessed/js/organisms/Header.js [hunk 3]
```diff
@@ -49,26 +49,26 @@ function Header( element ) {
     // Semi-opaque overlay that shows over the content when the menu flies out.
     _overlay = overlay;
 
+    // Don't initialize the mega menu if it isn't on the page.
+    if ( _dom.classList.contains( `${ BASE_CLASS }__mega-menu` ) ) {
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/organisms/Header.js]: How about this instead:

```
// Don't initialize the mega menu if it isn't on the page.
    _globalSearch = new GlobalSearch( _dom );

    if ( _dom.classList.contains( `${ BASE_CLASS }__mega-menu` ) ) {
      _megaMenu = new MegaMenu( _dom );
      _megaMenu.addEventListener( 'rootExpandBegin', _megaMenuExpandBegin );
      _megaMenu.addEventListener( 'rootCollapseEnd', _megaMenuCollapseEnd );
      _megaMenu.init();

      // If we have a mega menu, it needs to be collapsed when search is expanded.
      _globalSearch.addEventListener( 'expandBegin', _megaMenu.collapse );
    }

    _globalSearch.init();
```

Comment (chosak) [reply cfgov/unprocessed/js/organisms/Header.js]: Thanks, that is nicer. Applied in 844a499.

**Pull request merged**